checkov/terraform/checks/resource/aws/EFSEncryptionEnabled.py

```diff
@@ -0,0 +1,25 @@
+from checkov.terraform.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_check import BaseResourceCheck
+
+
+class EFSEncryptionEnabled(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure EFS is securely encrypted"
+        id = "CKV_AWS_42"
+        supported_resources = ['aws_efs_file_system']
+        categories = [CheckCategories.ENCRYPTION]
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf):
+        """
+            Looks for encryption configuration at EFS:
+            https://www.terraform.io/docs/providers/aws/r/efs_file_system.html
+        :param conf: efs configuration
+        :return: <CheckResult>
+        """
+        if "encrypted" in conf.keys():
```

Comment:
@JamesWoolfenden The fact that the key `encrypted` exists isn't enough as `encrypted` can be set to false

Reply:
i agree. The problem i was having was that my efs was a module and the values got no expansion at all.

Reply:
Ive change the checks, one for the encrypted value and another so that kms_key_id is set. I split out the tests to separate files to cover this .